capella_model_explorer/__main__.py

```diff
@@ -209,124 +205,160 @@ def build_image():
+    envvar="CME_HOST",
+    default=c.Defaults.host,
+    help=f"The hostname or IP address to bind to. Default: {c.Defaults.host}. "
+    "Ignored when running with '--container'.",
+)
+@click.option(
+    "--port",
+    envvar="CME_PORT",
+    default=c.Defaults.port,
+    help=f"The port to listen on. Default: {c.Defaults.port}.",
+)
+@click.option(
+    "--model",
+    envvar="CME_MODEL",
+    default=c.Defaults.model,
+    help="The Capella model to load (file, URL or JSON string). "
+    f"Default: {c.Defaults.model}.",
+)
+@click.option(
+    "--templates-dir",
+    envvar="CME_TEMPLATES_DIR",
+    default=str(c.Defaults.templates_dir.resolve()),
+    help="The directory containing the templates. "
+    f"Default: {c.Defaults.templates_dir.resolve()}.",
+)
+@click.option(
+    "--live-mode",
+    envvar="CME_LIVE_MODE",
+    default="1" if c.Defaults.live_mode else "0",
+    type=click.Choice(["0", "1"]),
+    help="Control automatic reloading of templates on changes. "
+    "Set to 1 to enable, 0 to disable. "
+    f"Default: {'1' if c.Defaults.live_mode else '0'}. "
+    "Ignored in '--dev' mode.",
+)
+@click.option(
+    "--route-prefix",
+    envvar="CME_ROUTE_PREFIX",
+    default="",
+    help="Add a prefix to all web routes. "
+    "(Note: this prefix does not apply to '/metrics').",
+)
+@click.option(
+    "--docker-image-name",
+    envvar="CME_DOCKER_IMAGE_NAME",
+    default=c.Defaults.docker_image_name,
+    help="The Docker image to use with '--container'. "
+    f"Default: {c.Defaults.docker_image_name}.",
+)
+def run(
+    container: bool,  # noqa: FBT001
+    dev: bool,  # noqa: FBT001
+    host: str,
+    port: int,
+    model: str,
+    templates_dir: str,
+    live_mode: str,
+    route_prefix: str,
+    docker_image_name: str,
+) -> None:
+    """Run the application."""
+    os.environ["CME_HOST"] = host
+    os.environ["CME_PORT"] = str(port)
+    os.environ["CME_MODEL"] = model
+    os.environ["CME_TEMPLATES_DIR"] = templates_dir
+    os.environ["CME_LIVE_MODE"] = live_mode
+    os.environ["CME_ROUTE_PREFIX"] = route_prefix
+    os.environ["CME_DOCKER_IMAGE_NAME"] = docker_image_name
+    importlib.reload(c)
+
+    if container and dev:
+        raise click.UsageError(
+            "Options --container and --dev are mutually exclusive."
+        )
+    if container:
+        run_container()
+    elif dev:
+        run_local_dev()
+    else:
+        run_local()
+
+
+@main.command()
+@click.option(
+    "--css",
+    "css_flag",
+    is_flag=True,
+    help="Build the style sheet with Tailwind CSS.",
+)
+@click.option(
+    "--container-image",
+    "container_image",
+    is_flag=True,
+    help="Build the Docker container image.",
+)
+def build(css_flag: bool, container_image: bool) -> None:  # noqa: FBT001
+    """Run a specified build process."""
+    if css_flag and container_image:
+        raise click.UsageError(
+            "Options --css and --container-image are mutually exclusive."
         )
```

Comment:
When we make the build target a required argument we can remove this usage check.